pyramid/router.py

```diff
@@ -213,7 +214,7 @@ def invoke_subrequest(self, request, use_tweens=False):
             try:
                 extensions = self.request_extensions
                 if extensions is not None:
-                    request._set_extensions(extensions)
+                    apply_request_extensions(request, extensions=extensions)
```

Comment:
This is just an optimization to have the extensions queried from the registry only once at app-setup-time. This code is in pyramid's hot route, so I wanted to keep it as close to the original version as possible.